packages/remote-config/android/src/main/java/io/invertase/firebase/config/UniversalFirebaseConfigModule.java

```diff
@@ -109,12 +108,15 @@
       }
 
       if (xmlResourceParser != null) {
-        FirebaseRemoteConfig.getInstance(firebaseApp).setDefaults(resourceId);
-        return null;
+        // REVIEW this changed from setDefaults to setDefaultsAsync returning Task<Void> -
+        // so adjusted the whole method to just return from this call or do previous behavior
+        // with Task based exception handling if resource not found. ?
+        return FirebaseRemoteConfig.getInstance(firebaseApp).setDefaultsAsync(resourceId);
```

Comment:
If you wrap this entire method back in a `return Tasks.call(getExecutor(), () -> {` as on line 101 then you can for this line just do:

```java
// ...

if (xmlResourceParser != null) {
  Tasks.await(
    FirebaseRemoteConfig.getInstance(firebaseApp).setDefaultsAsync(resourceId)
  );

  return null;
}

throw new Exception("resource_not_found");

// ...
```

This will wait for the task to execute and then continue execution - await is fine to do when wrapped with a `Tasks.call` as was previously (as it's on a different thread).